src/main.ts

```diff
@@ -3,6 +3,15 @@ import * as cache from './cache';
 
 export async function run() {
   try {
+    core.info('------------------------');
+    core.info(
+      'NOTE: this action is deprecated and will not be maintained in future by GitHub engineers.'
```

Comment:
This is almost certainly a regional thing, but "by GitHub engineers" doesn't sound very natural to me. I think if you just made it "by GitHub", it'd sound more natural and be more definitive. But going one step further, I don't think it really matters who is no longer maintaining it. By saying it's not maintained by X, you leave open the question about whether it could be maintained by Y. That isn't an option here. I'd suggest simplifying to:

> NOTE: This action is deprecated and is no longer maintained.